CC-Bootloader/rfcat_bootloader

```diff
@@ -55,82 +57,87 @@ def verify_code(ihx_file, serial_port):
           else:
             break
         if not can_read_any:
-          print "*** warning! this version of CC-Bootloader can only read 16 byte blocks!"
-          print "*** upgrade recommended!"
+          print("*** warning! this version of CC-Bootloader can only read 16 byte blocks!")
+          print("*** upgrade recommended!")
       if can_read_any:
-        block_length= length
+        block_length = length
       else:
-        block_length= ((length / 16) + 1) * 16
-      print "\rVerifying %04d bytes at address: %04X" % (length, start_addr),
+        block_length = ((length / 16) + 1) * 16
+      print("\rVerifying %04d bytes at address: %04X".format(length, start_addr))
       do_flash_read(serial_port, start_addr, block_length)
-      verify_data= ''
+      verify_data = ''
       for read_data in serial_port:
-        read_data= read_data.strip()
-        if (not data or read_data == ":00000001FF"):
+        read_data = read_data.strip()
+        if not data or read_data == ":00000001FF":
             break
         # strip header and checksum
         verify_data += read_data[9:-2]
-      if (data == verify_data[:length*2]):
-        print '(OK)',
+      if data == verify_data[:length * 2]:
+        print('(OK) ')
       else:
-        print 'Failed! Expected:', data, 'Got:', verify_data[:length*2]
+        print('Failed! Expected: ' + str(data) + ' Got: ' + str(verify_data[:length*2]))
         exit(1)
       sys.stdout.flush()
     else:
-      print "Skipping non data record: '%s'" % line[:-1]
+      print("Skipping non data record: '{}'".format(line[:-1]))
   return True
 
+
 def run_user_code(serial_port):
   # User code is entered on intel HEX EOF record
   serial_port.write(":00000001FF\n")
   return True
-  
+
+
 def reset_bootloader(serial_port):
   serial_port.write(":00000022DE\n")
   rc = serial_port.read()
-  print "RC =", rc,
+  print("RC = {}".format(rc))
   if rc in bootloader_error_codes:
-    print "(%s)" % bootloader_error_codes[rc]
+    print("(%s)".format(bootloader_error_codes[rc]))
   else:
-    print "(Unknown Error)"
-  if (rc != '0'):
-    print "Error resetting bootloader!"
+    print("(Unknown Error)")
+  if rc != '0':
+    print("Error resetting bootloader!")
     return False
   return True
 
+
 def erase_all_user(serial_port):
   serial_port.write(":00000023DD\n")
   rc = serial_port.read()
-  print "RC =", rc,
+  print("RC = ".format(rc))
   if rc in bootloader_error_codes:
-    print "(%s)" % bootloader_error_codes[rc]
+    print("(%s)".format(bootloader_error_codes[rc]))
```

Comment:
formatting error.  tell you what, before i finish this, why don't you take a stab through looking for these formatting errors and correcting them?

Reply:
Yeah doing that now. I had many late nights going through this after work